readthedocs/doc_builder/python_environments.py

```diff
@@ -430,6 +436,33 @@ def install_requirements_file(self, install):
                 bin_path=self.venv_bin(),
             )
 
+    def install_packages_list(self, install):
+        """
+        Install requirements from a string specification using pip.
+
+        :param install: A instal object from the config module.
+        :type install: readthedocs.config.modules.PythonInstallPackages
+        """
+
+        args = [
+            self.venv_bin(filename='python'),
+            '-m',
+            'pip',
+            'install',
+        ]
+        if self.project.has_feature(Feature.PIP_ALWAYS_UPGRADE):
+            args += ['--upgrade']
+        args += [
+            '--exists-action=w',
+            *self._pip_cache_cmd_argument(),
+        ]
+        args += install.packages
```

Comment:
We should be careful here, users could set a package name like `--option` and we will execute that. Not sure the best way to prevent that. We could probably validate this in the config module and raise an exception for invalid names, or maybe escape those characters here. /cc @readthedocs/core 

Invalid names would be those that start with `-`, not sure what else we should validate here.

Reply:
How much of a concern is this really? Pip also reads options from `requirements.txt` files, or at least a subset of options that are valid for the `pip install` command like 

```
--extra-index-url https://example.org/pypi/simple
some-requirement >=1.0.0, <2.0.0
```

Reply:
yeah, you are right, maybe allow the same subset of options?

Reply:
what about using the same parser that pip uses and check for a valid name?

On the other hand, @davidfischer created this package that could be useful if we can validate each line separately: https://requirements-parser.readthedocs.io/en/latest/

Reply:
Does RTD have a separate validation step? Otherwise Pip will just complain if the requirement is invalid.

Reply:
Right, but we need to verify it before sending to pip in some way because otherwise you can inject random commands into the `pip` by calling `pip install package1 "echo test > /tmp/test.txt"` (quotes should be `, but I can escape them here)

Reply:
> On the other hand, @davidfischer created this package that could be useful if we can validate each line separately: https://requirements-parser.readthedocs.io/en/latest/

This package is super old and outdated and probably would need some love before it's useful. That package doesn't directly parse the lines by itself but instead uses the parser built-in to setuptools.